### PR TITLE
Add auto-release

### PR DESCRIPTION
**Pull request by damassi**
Adds [auto-release](https://github.com/intuit/auto-release) as a replacement for `semantic-release`. 

(Will remove semantic-release in a follow-up PR.)

Related: 
- https://github.com/artsy/reaction/pull/1779

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit c9c11c911ca81119c898d4cd26c664ad93609ab9

https://deploy-preview-146--artsy-palette.netlify.com

File: .autorc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "no-version-prefix": true,
+  "onlyPublishWithReleaseLabel": true,
```
Comment (orta) [inline .autorc]: We'll need to remove this line

File: package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "prettier-project": "yarn prettier-write 'src/**/*.{ts,tsx}'",
     "prettier-write": "yarn prettier --write",
     "prettier": "prettier",
+    "release": "auto shipit",
```
Comment (orta) [inline package.json]: This will need calling in CI, but structurally this is sound 👍

Comment (damassi) [reply package.json]: Yup, i was going to do a second pass and remove the semantic-release flow in favor of this

**Comment by zephraph** [comment]
Hi friends. I'm in the neighborhood and figured I'd push these through. Hope you don't mind! ❤️

**Pull request merged**